torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -206,11 +206,13 @@ def use_foreach_copy(fsdp_param: FSDPParam) -> bool:
     for i, fsdp_param in enumerate(fsdp_params):
         if use_foreach_copy(fsdp_param):
             foreach_copy_indices.append(i)
-            all_gather_input = (
-                fsdp_param._sharded_param_data
-                if fsdp_param.sharded_state == ShardedState.SHARDED
-                else cast(torch.Tensor, fsdp_param._sharded_post_forward_param_data)
-            )
+            if fsdp_param.sharded_state == ShardedState.SHARDED:
+                all_gather_input = fsdp_param._sharded_param_data
+            else:
+                all_gather_input = cast(
+                    torch.Tensor, fsdp_param._sharded_post_forward_param_data
+                )
+            assert all_gather_input.ndim == 1, f"{all_gather_input.shape=}"
```

Comment:
just a refactor to be able to assert on flat input?

Reply:
yep -- I can revert this change

For uneven `Shard(i)`, I had to add an `elif` here, but I removed it now.